### PR TITLE
Dependency fix

### PR DESCRIPTION
**Pull request by 19majkel94**
This PR extract unnecessary deps to peer dependencies.

Closing #47.

### CODE REVIEW

**Comment by 19majkel94** [comment]
I will fix the metadata error later today 😉

**Comment by szokodiakos** [comment]
Hello. Unfortunately the test script change runs 3 tests only.

**Comment by 19majkel94** [comment]
Ok, now it should works on both linux/osx/travis and local windows :wink:

File: README.md [hunk 1]
```diff
@@ -112,10 +112,27 @@ Please note that sub documents doesn't have to extend Typegoose. You can still g
 
 `npm install typegoose -S`
 
+## Peer dependencies
+
+In order to make this lib works, you need to install also the peer dependencies:
```
Comment (szokodiakos) [inline README.md]: should be better: `In order to make this lib work, you also need to install the following peer dependencies:`

File: README.md [hunk 2]
```diff
@@ -112,10 +112,27 @@ Please note that sub documents doesn't have to extend Typegoose. You can still g
...
+
+* mongoose (`npm install mongoose -S`)
+* mongoose typings (`npm install @types/mongoose -S`)
+
+Also, you need to provide o polyfill for Metadata Reflection API for ECMAScript - it might be `core-js/es7/reflect` or [`reflect-metadata`](https://github.com/rbuckton/reflect-metadata). Remember that you have to import it at the top of your app entry file.
```
Comment (szokodiakos) [inline README.md]: typo `o polyfill`

**Comment by szokodiakos** [comment]
Thanks for fixing the test problem. The PR looks good, however I've commented on the docs. Also I'm thinking that this would introduce breaking change, as users must ship their own polyfills, is that correct?

**Comment by 19majkel94** [comment]
> Also I'm thinking that this would introduce breaking change, as users must ship their own polyfills, is that correct?

Right, so I've change the merge base to feature/4 😉

**Comment by rasdaniil** [comment]
Very much needed!

**Comment by 19majkel94** [comment]
Closing this as it's a bit outdated.

**Pull request closed without merging**